### DISCUSSION TITLE
what is purpose of checkStopShipXmls task?

### DISCUSSION THREAD

**Discussion by miduch** [xmartlabs/gong#273] [General]
checkStopShipXmls

https://github.com/xmartlabs/gong/blob/main/scripts/stop_ship_check_xmls.gradle.kts

**Comment by mirland** [comment]
Hey, thanks for asking. It's a linter that checks forbidden comments in XML files. For example, detekt has some [forbidden comments](https://github.com/xmartlabs/gong/blob/main/.detekt/detekt-config.yml#L615), so if you run the linter and you add a forbidden comment the linter will fail. However, detekt only runs in kt files, so we added a similar check in XML files. Right now XML files are not used a lot, because we're using compose, it's only used for strings. So I think we can remove it in the future.

BTW: the forbidden comments are used to fail the linter, for example, you add something just to test something and you have to delete it before submitting the PR, the forbidden comments will help you to don't forget change that code
